src/v/transform/api.cc

```diff
@@ -131,21 +138,38 @@ class partition_source final : public source {
         // of the log.
         model::offset start_offset = std::max(
           result.value(), kafka::offset_cast(offset));
+        model::offset max_offset = model::offset::max();
+        // If transactions are enabled, clamp reads to only completed
```

Comment:
shouldn't we always clamp to lso? (not just when transactions are enabled)

Reply:
Yeah we do this, I just need to fix the comment.